packages/compass-components/src/components/interactive-popover.tsx

```diff
@@ -148,7 +165,7 @@ function InteractivePopover({
   const closeButtonId = useId('close-button-id');
 
   return trigger({
-    onClick: onClickTrigger,
+    onClick: onClickTrigger as any,
```

Comment:
There is almost never a case where you need to asser any for a good reason, the type of the event argument in this onClickTrigger method should be `React.MouseEventHandler<HTMLButtonElement>`

Reply:
It could be also a React.TouchEventHandler, if you see the signature of onClickTrigger, it's (event: MouseEvent | TouchEvent) => void;

Unless there is a way to specify that is both of them (or a common parent), it would require changing the typing of all components using the popover, which then I would suggest to have a specific task to clean up the whole InteractivePopover as there are already a few things that might need a refactor.

Reply:
I'll be using this type:

`const onClickTrigger: React.EventHandler<React.MouseEvent<HTMLButtonElement> | React.TouchEvent<HTMLButtonElement>> = useCallback(
    (event: React.MouseEvent<HTMLButtonElement> | React.TouchEvent<HTMLButtonElement>) => void
  );`

It's a bit more complex, but doesn't require the any.

Reply:
How is the TouchEvent relevant here? Both browser events and React onClick callbacks will always provide a MouseEvent instance on click